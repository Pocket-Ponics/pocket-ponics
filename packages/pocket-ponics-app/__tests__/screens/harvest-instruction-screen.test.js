import React from 'react'
import { shallow } from 'enzyme'
import { TouchableOpacity } from 'react-native'

import HarvestInstructionScreen from '../../screens/harvest-instruction-screen'

describe('Harvest Instruction Screen', () => {
	test('renders correctly', () => {
		global.plants = []
		const tree = shallow(
			<HarvestInstructionScreen  navigation={{ 
				getParam: jest.fn().mockReturnValue(5)
			}}/>
		).html()
		expect(tree).toMatchSnapshot()
	})
	test('renders greenbean', () => {
		global.plants = []
		const tree = shallow(
			<HarvestInstructionScreen  navigation={{ 
				getParam: jest.fn().mockReturnValue(2)
			}}/>
		).html()
		expect(tree).toMatchSnapshot()
	})
	test('renders spinach', () => {
		global.plants = []
		const tree = shallow(
			<HarvestInstructionScreen  navigation={{ 
				getParam: jest.fn().mockReturnValue(3)
			}}/>
		).html()
		expect(tree).toMatchSnapshot()
	})
	test('renders turnip', () => {
		global.plants = []
		const tree = shallow(
			<HarvestInstructionScreen  navigation={{ 
				getParam: jest.fn().mockReturnValue(4)
			}}/>
		).html()
		expect(tree).toMatchSnapshot()
	})

	test('goes to next screen', () => {
		global.plants = []
		const tree = shallow(
			<HarvestInstructionScreen  navigation={{ 
				getParam: jest.fn().mockReturnValue(3),
				navigate: jest.fn(),
				state: { params: {} }
			}}/>
		)
		tree.find(TouchableOpacity).at(0).props().onPress()
		expect(tree.html()).toMatchSnapshot()
	})

	test('goes to next screen', () => {
		global.plants = []
		const tree = shallow(
			<HarvestInstructionScreen  navigation={{ 
				getParam: jest.fn().mockReturnValue(4),
				navigate: jest.fn(),
				state: { params: {} }
			}}/>
		)
		tree.find(TouchableOpacity).at(0).props().onPress()
		expect(tree.html()).toMatchSnapshot()
	})
})