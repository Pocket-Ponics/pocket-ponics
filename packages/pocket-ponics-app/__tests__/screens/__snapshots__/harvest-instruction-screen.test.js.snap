// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Harvest Instruction Screen goes to next screen 1`] = `"<View style=\\"flex:1\\"><View style=\\"background-color:#472600;color:#FFFFFF;padding:20px;flex-direction:column;flex:1\\"><Text style=\\"color:#FFFFFF;font-size:32px;margin-bottom:20px;text-align:center\\">Harvest</Text><Text style=\\"color:#FFFFFF;font-size:18px;margin-bottom:20px\\"></Text><Image source=\\"1\\" style=\\"resize-mode:contain;width:100%;flex:1;margin-bottom:20px\\"></Image><View style=\\"background-color:#456E31;border-radius:5px;padding:10px;margin:10px;align-items:center;opacity:1\\"><Text style=\\"color:#FFFFFF;font-size:22px\\">Done Harvesting</Text></View></View></View>"`;

exports[`Harvest Instruction Screen goes to next screen 2`] = `"<View style=\\"flex:1\\"><View style=\\"background-color:#472600;color:#FFFFFF;padding:20px;flex-direction:column;flex:1\\"><Text style=\\"color:#FFFFFF;font-size:32px;margin-bottom:20px;text-align:center\\">Harvest</Text><Text style=\\"color:#FFFFFF;font-size:18px;margin-bottom:20px\\"></Text><Image source=\\"1\\" style=\\"resize-mode:contain;width:100%;flex:1;margin-bottom:20px\\"></Image><View style=\\"background-color:#456E31;border-radius:5px;padding:10px;margin:10px;align-items:center;opacity:1\\"><Text style=\\"color:#FFFFFF;font-size:22px\\">Done Harvesting</Text></View></View></View>"`;

exports[`Harvest Instruction Screen renders correctly 1`] = `"<View style=\\"flex:1\\"><View style=\\"background-color:#472600;color:#FFFFFF;padding:20px;flex-direction:column;flex:1\\"><Text style=\\"color:#FFFFFF;font-size:32px;margin-bottom:20px;text-align:center\\">Harvest</Text><Text style=\\"color:#FFFFFF;font-size:18px;margin-bottom:20px\\"></Text><Image source=\\"1\\" style=\\"resize-mode:contain;width:100%;flex:1;margin-bottom:20px\\"></Image><View style=\\"background-color:#456E31;border-radius:5px;padding:10px;margin:10px;align-items:center;opacity:1\\"><Text style=\\"color:#FFFFFF;font-size:22px\\">Done Harvesting</Text></View></View></View>"`;

exports[`Harvest Instruction Screen renders greenbean 1`] = `"<View style=\\"flex:1\\"><View style=\\"background-color:#472600;color:#FFFFFF;padding:20px;flex-direction:column;flex:1\\"><Text style=\\"color:#FFFFFF;font-size:32px;margin-bottom:20px;text-align:center\\">Harvest</Text><Text style=\\"color:#FFFFFF;font-size:18px;margin-bottom:20px\\"></Text><Image source=\\"1\\" style=\\"resize-mode:contain;width:100%;flex:1;margin-bottom:20px\\"></Image><View style=\\"background-color:#456E31;border-radius:5px;padding:10px;margin:10px;align-items:center;opacity:1\\"><Text style=\\"color:#FFFFFF;font-size:22px\\">Done Harvesting</Text></View></View></View>"`;

exports[`Harvest Instruction Screen renders spinach 1`] = `"<View style=\\"flex:1\\"><View style=\\"background-color:#472600;color:#FFFFFF;padding:20px;flex-direction:column;flex:1\\"><Text style=\\"color:#FFFFFF;font-size:32px;margin-bottom:20px;text-align:center\\">Harvest</Text><Text style=\\"color:#FFFFFF;font-size:18px;margin-bottom:20px\\"></Text><Image source=\\"1\\" style=\\"resize-mode:contain;width:100%;flex:1;margin-bottom:20px\\"></Image><View style=\\"background-color:#456E31;border-radius:5px;padding:10px;margin:10px;align-items:center;opacity:1\\"><Text style=\\"color:#FFFFFF;font-size:22px\\">Done Harvesting</Text></View></View></View>"`;

exports[`Harvest Instruction Screen renders turnip 1`] = `"<View style=\\"flex:1\\"><View style=\\"background-color:#472600;color:#FFFFFF;padding:20px;flex-direction:column;flex:1\\"><Text style=\\"color:#FFFFFF;font-size:32px;margin-bottom:20px;text-align:center\\">Harvest</Text><Text style=\\"color:#FFFFFF;font-size:18px;margin-bottom:20px\\"></Text><Image source=\\"1\\" style=\\"resize-mode:contain;width:100%;flex:1;margin-bottom:20px\\"></Image><View style=\\"background-color:#456E31;border-radius:5px;padding:10px;margin:10px;align-items:center;opacity:1\\"><Text style=\\"color:#FFFFFF;font-size:22px\\">Done Harvesting</Text></View></View></View>"`;
